src/tests/conformance/handshake.rs

```diff
@@ -343,6 +343,79 @@ async fn reject_version_reusing_nonce() {
     node.stop().await;
 }
 
+#[tokio::test]
+async fn reject_obsolete_versions() {
+    // ZG-CONFORMANCE-007
+    //
+    // The node rejects connections with obsolete node versions.
+    //
+    // We expect the following behaviour, regardless of who initiates the connection:
+    //
+    //  1. We send `version` with an obsolete version number
+    //  2. The node responds with `Reject(Obsolete)`
+    //  3. The node terminates the connection
+    //
+    // Neither Zebra nor ZCashd currently fully comply with this behaviour, so we may need
```

Comment:
Could we leave a clear comment that this test is expected to fail in its current state? Or maybe a `TODO: confirm expected behaviour.` at the top just so we have a consistent way of keeping track of these?

Reply:
Updated with additional comments